source/structured-streaming.txt

```diff
@@ -0,0 +1,327 @@
+
+     - id: python
+       content: |
+
+         .. code-block:: python
+            :copyable: true
+            :emphasize-lines: 3, 9
+         
+            streamingDataFrame = (<local SparkSession>
+              .readStream
+              .format("mongodb")
+              .load()
+            )
+         
+            query = (streamingDataFrame
+              .writeStream
+              .trigger(continuous="1 second")
+              .format("memory")
+              .outputMode("append")
+            )
+
+            query.start()
+
+         For a complete list of methods, see the 
+         `pyspark Structured Streaming reference <https://spark.apache.org/docs/latest/api/python/reference/pyspark.ss.html>`__.
+
+     - id: scala
+       content: |
+
+Examples
+--------
+
+The following examples show Spark Structured Streaming configurations 
+for streaming between MongoDB and a CSV file.
+
+Stream to MongoDB from a CSV File
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. tabs-drivers::
+
+   tabs:
+     - id: java-sync
+       content: |
+
+         .. code-block:: java
+            :copyable: true
+
+     - id: python
+       content: |
+
+         To create a :ref:`write stream <write-structured-stream>` to 
+         MongoDB from a ``.csv`` file, first create a `DataStreamReader <https://spark.apache.org/docs/latest/api/python/reference/api/pyspark.sql.streaming.DataStreamReader.html>`__ 
+         from the ``.csv`` file, then use that ``DataStreamReader`` to 
+         create a `DataStreamWriter <https://spark.apache.org/docs/latest/api/python/reference/api/pyspark.sql.streaming.DataStreamWriter.html>`__ 
+         to MongoDB. Finally, use the ``.start()`` method to begin the 
+         stream.
+         
+         As streaming data is read from the ``.csv`` file, it is added 
+         to MongoDB in the `outputMode <https://spark.apache.org/docs/latest/api/python/reference/api/pyspark.sql.streaming.DataStreamWriter.outputMode.html#pyspark.sql.streaming.DataStreamWriter.outputMode>`__ 
+         you specify.
+
+         .. code-block:: python
+            :copyable: true
+            :emphasize-lines: 18, 24
+
+            # create a local SparkSession
+            spark = SparkSession \
+              .builder \
+              .appName("writeExample") \
+              .master("spark://spark-master:<port>") \
+              .config("spark.jars", "<mongodb-spark-connector-{+current-version+}>.jar") \
+              .getOrCreate()
+ 
+            # define the schema of the .csv
+            readSchema = (StructType()
+              .add('Type', StringType())
+              .add('Date', TimestampType())
+              .add('Price', DoubleType())
+            )
+
+            # define a streaming query
+            query = (spark
+              .readStream
+              .format("csv")
+              .option("header", "true")
+              .schema(readSchema)
+              .load("daily*.csv")
+              # manipulate your streaming data
```

Comment:
Issue: As this no longer corresponds to any specific CSV file, could the content of the code snippet corresponding the a specific CSV file be removed? If accepted, this also applies to the following example on the page.

```suggestion
            # define the schema of the .csv
            readSchema = (<csv file schema>)

            # define a streaming query
            query = (spark
              .readStream
              .format("csv")
              .option("header", "true")
              .schema(readSchema)
              .load("<file name>.csv")
              # manipulate your streaming data
```

Reply:
Good call

Reply:
Simplified a little further to

            query = (spark
              .readStream
              .format("csv")
              .option("header", "true")
              .schema(<csv-schema>)
              .load(<csv-file-name>)